guides/source/getting_started.md

```diff
@@ -973,521 +792,436 @@ class ArticlesController < ApplicationController
+
+  <div>
+    <label for="article_title">Title</label><br>
+    <input type="text" name="article[title]" id="article_title">
+  </div>
+
+  <div>
+    <label for="article_body">Body</label><br>
+    <textarea name="article[body]" id="article_body"></textarea>
+  </div>
+
+  <div>
+    <input type="submit" name="commit" value="Create Article" data-disable-with="Create Article">
+  </div>
+</form>
+```
+
+TIP: To learn more about form builders, see [Action View Form Helpers](
+form_helpers.html).
+
+#### Using Strong Parameters
+
+Submitted form data is put into the `params` Hash, alongside captured route
+parameters. Thus, the `create` action can access the submitted title via
+`params[:article][:title]` and the submitted body via `params[:article][:body]`.
+We could pass these values individually to `Article.new`, but that would be
+verbose and possibly error-prone. And it would become worse as we add more
+fields.
+
+Instead, we will pass a single Hash that contains the values. However, we must
+still specify what values are allowed in that Hash. Otherwise, a malicious user
+could potentially submit extra form fields and overwrite private data. In fact,
+if we pass the unfiltered `params[:article]` Hash directly to `Article.new`,
+Rails will raise a `ForbiddenAttributesError` to alert us about the problem.
+So we will use a feature of Rails called *Strong Parameters* to filter `params`.
+Think of it as [strong typing](https://en.wikipedia.org/wiki/Strong_and_weak_typing)
+for `params`.
+
+Let's add a private method to the bottom of `app/controllers/articles_controller.rb`
+named `article_params` that filters `params`. And let's change `create` to use
+it:
 
-### Adding Some Validation
+```ruby
+class ArticlesController < ApplicationController
+  def index
+    @articles = Article.all
+  end
 
-The model file, `app/models/article.rb` is about as simple as it can get:
+  def show
+    @article = Article.find(params[:id])
+  end
 
-```ruby
-class Article < ApplicationRecord
-end
-```
+  def new
+    @article = Article.new
+  end
 
-There isn't much to this file - but note that the `Article` class inherits from
-`ApplicationRecord`. `ApplicationRecord` inherits from `ActiveRecord::Base`
-which supplies a great deal of functionality to your Rails models for free,
-including basic database CRUD (Create, Read, Update, Destroy) operations, data
-validation, as well as sophisticated search support and the ability to relate
-multiple models to one another.
+  def create
+    @article = Article.new(article_params)
 
-Rails includes methods to help you validate the data that you send to models.
-Open the `app/models/article.rb` file and edit it:
+    if @article.save
+      redirect_to @article
+    else
+      render :new
+    end
+  end
 
-```ruby
-class Article < ApplicationRecord
-  validates :title, presence: true,
-                    length: { minimum: 5 }
+  private
+    def article_params
+      params.require(:article).permit(:title, :text)
```

Comment:
```suggestion
      params.require(:article).permit(:title, :body)
```

Reply:
Good catch!  I forgot to change this when I copied and pasted from the old code examples.  I've made sure all occurrences of `article_params` use `:body` now.